src/Gui/BaseApplication.cpp

```diff
@@ -498,6 +498,9 @@ void BaseApplication::radiumFrame() {
 
     timerData.tasksEnd = Core::Utils::Clock::now();
 
+    // Update texture (cpu->gpu)
+    m_engine->getTextureManager()->updatePendingTextures();
```

Comment:
Use this line after startRendering call, your BaseApplication works for my application, but not for your example